crates/storage/store_db/rocksdb.rs

```diff
@@ -607,13 +671,93 @@ impl Store {
             };
         }
     }
+
+    fn apply_trie_updates(
+        &self,
+        notify: SyncSender<Result<(), StoreError>>,
+        parent_state_root: H256,
+        child_state_root: H256,
+        account_updates: Vec<(Nibbles, Vec<u8>)>,
+        storage_updates: StorageUpdates,
+    ) -> Result<(), StoreError> {
+        let db = &*self.db;
+        let fkv_ctl = &self.flatkeyvalue_control_tx;
+        let trie_cache = &self.trie_cache;
+
+        // Phase 1: update the in-memory diff-layers only, then notify block production.
+        let new_layer = storage_updates
+            .into_iter()
+            .flat_map(|(account_hash, nodes)| {
+                nodes
+                    .into_iter()
+                    .map(move |(path, node)| (apply_prefix(Some(account_hash), path), node))
+            })
+            .chain(account_updates)
+            .collect();
+        // Read-Copy-Update the trie cache with a new layer.
+        let trie = trie_cache
+            .lock()
+            .map_err(|_| StoreError::LockError)?
+            .clone();
+        let mut trie_mut = (*trie).clone();
+        trie_mut.put_batch(parent_state_root, child_state_root, new_layer);
+        *trie_cache.lock().map_err(|_| StoreError::LockError)? = Arc::new(trie_mut);
+        // Update finished, signal block production.
+        notify.send(Ok(())).map_err(|_| StoreError::LockError)?;
+
+        // Phase 2: update disk layer.
+        let trie = trie_cache
+            .lock()
+            .map_err(|_| StoreError::LockError)?
+            .clone();
```

Comment:
I think here we can clone the `Arc::new(trie_mut)` in line 704 before storing it in the cache, which would avoid this lock here.